### DISCUSSION TITLE
Component testing in Nuxt : "n is not a function" when including plugins

### DISCUSSION THREAD

**Discussion by franck-grenier** [cypress-io/cypress#30397] [Questions and Help]
Hello Cypress,
I'm testing my Nuxt Vue components with Cypress and I sometimes need to include plugins when I mount my component.

I do the following: 

```javascript
import UserAvatar from "@components/UserAvatar.vue";
import placeholder from '@plugins/placeholder.js';

describe('UserAvatar.cy.js', () => {
    it('should display placeholder if told so', () => {
        cy.mount(UserAvatar, {
            propsData: {
                user: {
                    "firstName": "Jeanne",
                    "lastName": "Alouin",
                    "profilePictureUrl": "https://somewhereontheweb.com",
                },
                usePlaceholder: true
            },
            extensions: {
                plugins: [placeholder],
            },
        })
        cy.get('[data-test="user-avatar_img"]').should('have.attr', 'src', 'https://somewhereontheweb.com').not()
        cy.get('[data-test="user-avatar_img"]').should('have.attr', 'alt', 'Photo de profil')
    })
})
```

My `placeholder` plugin is: 

```javascript
export default (context, inject) => {
    inject('placeholderProfilePicture', placeholderProfilePicture);
    inject('placeholderDocumentThumbnail', placeholderDocumentThumbnail);
};
```

When I include the plugin in the `mount` method call, I get an error on `inject` instructions : 

```
TypeError: t is not a function
```

It looks like the Nuxt `inject` function is unknown in the Cypress component testing context.

What should I do ? I can't find any documentation about this...

Thanks for your help

**Comment by portyu9** [comment]
Hi @franck-grenier 

`extensions.plugins` expects a **Vue plugin**. Your `placeholder` file is a **Nuxt plugin** whose signature is:

```js
(context, inject) => { ... }
```

Cypress's Vue component mount does not boot the Nuxt plugin runtime, so there is no Nuxt `inject` function for Cypress to pass there. That is why it fails when it reaches `inject(...)`.

For this component test, either:

- provide/mock the injected `$placeholder...` values at the Vue component boundary, or
- move `placeholderProfilePicture` / `placeholderDocumentThumbnail` into a plain module and import/inject those functions directly.

If the behavior you want to test specifically depends on Nuxt's plugin lifecycle, test it in a running Nuxt application rather than a standalone Vue component mount.

There is also an important current-version difference: Cypress now supports the Nuxt 3+/Vue 3 component-testing path. Nuxt 2/Vue 2 component testing is no longer supported out of the box; the old `@cypress/vue2` harness is deprecated and intended only as a temporary migration workaround.

So for a current project, the preferred long-term fix is Nuxt 3+/Vue 3 rather than building more Nuxt 2 CT infrastructure.

Please mark this answer as accepted if it helped, thank you.
